core/ports/services.go

```diff
@@ -40,21 +61,71 @@ type EventEnvelope struct {
 
 // ImageService is a port for a service that interacts with OCI images.
 type ImageService interface {
-	// Get will get (i.e. pull) the image for a specific owner.
-	Get(ctx context.Context, input GetImageInput) error
-	// GetAndMount will get (i.e. pull) the image for a specific owner and then
+	// Pull will get (i.e. pull) the image for a specific owner.
+	Pull(ctx context.Context, input *ImageSpec) error
+	// PullAndMount will get (i.e. pull) the image for a specific owner and then
 	// make it available via a mount point.
-	GetAndMount(ctx context.Context, input GetImageInput) ([]models.Mount, error)
+	PullAndMount(ctx context.Context, input *ImageMountSpec) ([]models.Mount, error)
```

Comment:
If it has a separate Pull, maybe a Mount function can be useful. For example if we want to control the full workflow and call `Pull`, verify, and then `Mount`. With this, we can call `Pull`, verify, and `PullAndMerge`, which is fine if it's guaranteed it will be the same and does not try to pull and/or check if the remote has been change again